text/0000-idempotent-install.md

```diff
@@ -0,0 +1,68 @@
+- Start Date: 2016-12-13
+- RFC PR: (leave this empty)
+- Yarn Issue: (leave this empty)
+
+# Summary
+
+`yarn install` should be idempotent.
+
+Ideally, the result of `yarn install` would not be statefully dependent on the contents of an existing `node_modules`, and would ensure the resulting `node_modules` is identical regardless of whether there is an existing `node_modules` or not. This seems very inline with the spirit of yarn's goal of deterministic builds (same `node_modules` independent of whether `node_modules` exists or the version of node that generated it).

```

Comment:
Thanks, the example makes it clear.

Integrity file is generated here https://github.com/yarnpkg/yarn/blob/ad11748d218d0d2d937ca4e739168660193168dc/src/cli/commands/install.js#L680 based on a few parameters: arguments, dependencies etc.
I think we can add a new parameter - `process.version`.
It would invalidate node_modules when you switch node version.